### PR TITLE
Temporary fix problems with Quantity.__class_getitem__

### PR DESCRIPTION
**Pull request by mhvk**
Not a permanent fix, but we need to get the tests running again!

EDIT: also include a fix for `Masked` no longer working with `np.quantile`.

### CODE REVIEW

**Comment by mhvk** [comment]
Hmm, looks like there are other numpy-dev problems too...  Sigh.

**Comment by nstarman** [comment]
I can take a look at this later tonight.

**Comment by mhvk** [comment]
OK, added a commit to fix the other numpy-dev problems...

**Comment by mhvk** [comment]
OK, looks like all tests now pass again, which will help unblock everything else. Given that, cc @pllim...

**Review by nstarman — APPROVED** [review]
The ``NDArray.__class_getitem__`` looks good to me.

**Comment by pllim** [comment]
Thanks! Is there a follow-up issue for a proper fix?

**Comment by pllim** [comment]
@meeseeksdev backport to v5.0.x

**Comment by pllim** [comment]
For future reference, when you intend to merge a PR milestoned for v5.0.x, please also add a `backport-v5.0.x` label *before* merge. Thanks!

**Pull request merged**
